integration_tests/src/main/python/delta_lake_write_test.py

```diff
@@ -54,7 +54,18 @@ def del_keys(key_list, c_val, g_val):
 
 def fixup_operation_metrics(opm):
     """Update the specified operationMetrics node to facilitate log comparisons"""
-    for k in "executionTimeMs", "numOutputBytes", "rewriteTimeMs", "scanTimeMs":
+    # note that we remove many byte metrics because number of bytes can vary
+    # between CPU and GPU. We skip `numTargetRows*` due to
+    # https://github.com/NVIDIA/spark-rapids/issues/8556 and this is currently
+    # safe because we fall back to CPU for queries involving
+    # `NOT MATCHED BY SOURCE. Tracking issue is https://github.com/NVIDIA/spark-rapids/issues/8415
+    for k in "executionTimeMs", "numOutputBytes", "rewriteTimeMs", "scanTimeMs", \
+            "numRemovedBytes", "numAddedBytes", "numTargetBytesAdded", "numTargetBytesInserted", \
+            "numTargetBytesUpdated", "numTargetBytesRemoved", \
+            'numTargetRowsMatchedUpdated', \
+            'numTargetRowsMatchedDeleted', \
+            'numTargetRowsNotMatchedBySourceUpdated', \
+            'numTargetRowsNotMatchedBySourceDeleted':
```

Comment:
for easier maintenance in the future, I would create a separate variable with the list of metrics one per line,
```Python 
metrics_to_remove = [
  'executionTimeMs', 
  'numOutputBytes', 
  ...
]
```

nit: I'd stick to some consistent use of quotes or double-quotes in a single code fragment at least